src/objc/abstract/MCOAbstractMessage.mm

```diff
@@ -30,9 +34,9 @@ @implementation MCOAbstractMessage {
 
 - (id) init
 {
-    self = [self initWithMCMessage:NULL];
     MCAssert(0);
-    return nil;
+    [self release];
```

Comment:
You can keep the previous code. MCAssert() is making sure that it's not used and crashing.